lib/addressable/uri.rb

```diff
@@ -1558,6 +1558,15 @@ def basename
       return File.basename(self.path).sub(/;[^\/]*$/, EMPTY_STR)
     end
 
+    ##
+    # Sets the basename component for this URI.
+    #
+    # @param [String, #to_str] new_basename. The new basename value
+    def basename=(new_basename)
+      self.path = path.sub(%r{#{basename}(\/)?\z}, new_basename)
```

Comment:
I'd kind of like this regexp to ensure that the `basename` occurs at a boundary. Theoretically the `File.basename` call already addresses this, but in many other scenarios where you interpolate into a regexp, I'd expect to find a bug, and the only reason this isn't one is because `File.basename` finds the boundary. Adding a `\b` makes it more obvious that there isn't a bug here.